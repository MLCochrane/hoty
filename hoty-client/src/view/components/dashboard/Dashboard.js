import React, { Component } from 'react';
import PageBar from '../global/header/PageBar';
import {
  socketChat,
} from '../../../api';
// import PropTypes from 'prop-types';

const chatInterface = socketChat();

class Dashboard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      message: '',
      msg: [],
    };

    chatInterface.recieve((id, msg) => {
      this.setState(prevState => ({
        msg: [...prevState.msg, { id, msg }],
      }));
    });

    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(e) {
    const { target } = e;
    const { value } = target;

    this.setState({
      message: value,
    });
  }

  render() {
    const {
      message,
      msg,
    } = this.state;
    return (
      <div className="dashboard">
        <PageBar
          title="Dashboard"
        />
        <textarea
          value={message}
          onChange={e => this.handleChange(e)}
        />
        <button
          type="button"
          onClick={() => { chatInterface.send(message); }}
        >
          click me
        </button>
        <div>
          <ul>
            {msg.map(el => (
              <li
                key={el.id}
              >
                {el.msg}
              </li>
            ))}
          </ul>
        </div>
      </div>
    );
  }
}

export default Dashboard;

// Dashboard.propTypes = {
// };
