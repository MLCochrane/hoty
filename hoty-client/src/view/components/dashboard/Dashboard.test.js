import React from 'react';
import { create } from 'react-test-renderer';
import Dashboard from './Dashboard';

// const Dashboard = () => (
//   <div className="dashboard">
//     <Toolbar
//       title="Dashboard"
//     />
//   </div>
// );
describe('Dashboard component', () => {
  test('dashboard renders', () => {
    const component = create(
      <Dashboard />,
    );
    expect(component).toMatchSnapshot();
  });
});
