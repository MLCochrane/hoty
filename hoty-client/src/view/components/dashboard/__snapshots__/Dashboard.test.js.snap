// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component dashboard renders 1`] = `
<div
  className="dashboard"
>
  <div
    className="makeStyles-bar-1"
  >
    <div
      className="makeStyles-wrap-2"
    >
      <h1
        className="MuiTypography-root makeStyles-title-3 MuiTypography-h2"
      >
        Dashboard
      </h1>
      <div
        className="page-bar__content"
      />
    </div>
  </div>
</div>
`;
